### PR TITLE
Move file reading docs to cairo

### PR DESCRIPTION
**Pull request by Arcticae**
## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: snforge_std/src/fs/file_operations.cairo [hunk 1]
```diff
@@ -16,6 +18,8 @@ impl FileTraitImpl of FileTrait {
     }
 }
 
+/// `file` - a `File` struct you want to read text data from
```
Comment (drknzz) [inline snforge_std/src/fs/file_operations.cairo:21]:
```suggestion
/// `file` - a `File` struct to read text data from
```

File: snforge_std/src/fs/file_operations.cairo [hunk 2]
```diff
@@ -32,6 +36,8 @@ fn read_txt(file: @File) -> Array<felt252> {
     result
 }
 
+/// `file` - a `File` struct you want to read text data from
```
Comment (drknzz) [inline snforge_std/src/fs/file_operations.cairo:39]:
```suggestion
/// `file` - a `File` struct to read json data from
```

File: snforge_std/src/fs/file_operations.cairo [hunk 3]
```diff
@@ -49,7 +55,13 @@ fn read_json(file: @File) -> Array<felt252> {
 }
 
 trait FileParser<T, impl TSerde: Serde<T>> {
+    /// Reads from the text file and tries to deserialize the result into the given type with `Serde`
+    /// `file` - file instance
+    /// Returns an instance of `T` if deserialization was possible
     fn parse_txt(file: @File) -> Option<T>;
+    /// Reads from the json file and tries to deserialize the result into the given type with `Serde`
+    /// `file` - file instance
+    /// Returns an instance of `T` if deserialization was possible
```
Comment (drknzz) [inline snforge_std/src/fs/file_operations.cairo:64]:
```suggestion
    /// Reads from the text file and tries to deserialize the result into given type with `Serde`
    /// `file` - File instance
    /// Returns an instance of `T` if deserialization was possible
    fn parse_txt(file: @File) -> Option<T>;
    /// Reads from the json file and tries to deserialize the result into given type with `Serde`
    /// `file` - File instance
    /// Returns an instance of `T` if deserialization was possible
```

File: snforge_std/src/fs/file_operations.cairo [hunk 4]
```diff
@@ -3,10 +3,12 @@ use super::super::byte_array::byte_array_as_felt_array;
 
 #[derive(Drop, Clone)]
 struct File {
-    path: ByteArray // relative path
+    path: ByteArray
 }
 
 trait FileTrait {
+    /// Creates a file struct which you can use for reading json or text
+    /// `path` - a path in ByteArray form, relative to the package root
```
Comment (drknzz) [inline snforge_std/src/fs/file_operations.cairo:11]:
```suggestion
    /// `path` - a path to file in ByteArray form, relative to the package root
```

**Review by drknzz — APPROVED** [review]
Do we want to keep the separate files? Wasn't it done a bit different in signature module docs revamp?

**Comment by Arcticae** [comment]
> Do we want to keep the separate files? Wasn't it done a bit different in signature module docs revamp?

Yeah you're probably right - i will merge those into one.

**Review by Draggu — APPROVED** [review]

File: docs/src/appendix/snforge-library/fs.md [hunk 5]
```diff
@@ -2,9 +2,91 @@
...
+```
+
+## `read_txt` & `read_json`
+```rust
+fn read_txt(file: @File) -> Array<felt252>;
+fn read_json(file: @File) -> Array<felt252>;
+```
+
+
+## File format
```
Comment (MaksymilianDemitraszek) [inline docs/src/appendix/snforge-library/fs.md:27]: As an end goal we probably would like to put those in cairo, yup?

Comment (Arcticae) [reply docs/src/appendix/snforge-library/fs.md:27]: I thought of a guide here actually, but both would work

File: docs/src/appendix/snforge-library/fs.md [hunk 6]
```diff
@@ -2,9 +2,91 @@
...
+## `File`
+```
+trait FileTrait {
+    fn new(path: ByteArray) -> File;
+}
+```
+
+## `FileParser<T>`
+```rust
+trait FileParser<T, impl TSerde: Serde<T>> {
```
Comment (piotmag769) [inline docs/src/appendix/snforge-library/fs.md:14]:
```suggestion
trait FileParser<T, +Serde<T>> {
```
https://github.com/foundry-rs/starknet-foundry/issues/2010

File: docs/src/appendix/snforge-library/fs.md [hunk 7]
```diff
@@ -2,9 +2,91 @@
...
+## `read_txt` & `read_json`
+```rust
+fn read_txt(file: @File) -> Array<felt252>;
+fn read_json(file: @File) -> Array<felt252>;
+```
+
+
+## File format
+Some rules have to be checked when providing a file for snforge, in order for correct parsing behavior.
+Different ones apply for JSON and plain text files.
```
Comment (piotmag769) [inline docs/src/appendix/snforge-library/fs.md:29]:
```suggestion
The rules differ between JSON and plain text files.
```

File: docs/src/appendix/snforge-library/fs.md [hunk 8]
```diff
@@ -2,9 +2,91 @@
...
+
+### JSON files
+- Elements have to be either:
+  - integers in range of `[0, P)` where P is [`Cairo Prime`](https://book.cairo-lang.org/ch02-02-data-types.html?highlight=prime#felt-type)
+  - single line strings (`ByteArray`) i.e. `"very very very very loooooong string"`, new lines can be used with `\n` and `"` with `\"`
+  - array of integers or strings fulfilling the above conditions
+
+> ⚠️ **Warning**
+>
+> A JSON object is an unordered data structure. To give it an order, the values in the array are sorted alphabetically by JSON keys. Therefore, the values in the array are sorted alphabetically by JSON key.
```
Comment (piotmag769) [inline docs/src/appendix/snforge-library/fs.md:47]:
```suggestion
> A JSON object is an unordered data structure. To make reading JSONs deterministic, the values are read from the JSON in an order that is alphabetical in respect to JSON keys.
```

File: snforge_std/src/fs/file_operations.cairo [hunk 9]
```diff
@@ -4,10 +4,12 @@ use super::super::_cheatcode::handle_cheatcode;
 
 #[derive(Drop, Clone)]
 struct File {
-    path: ByteArray // relative path
+    path: ByteArray
 }
 
 trait FileTrait {
+    /// Creates a file struct used for reading json / text
+    /// `path` - a path to file in ByteArray form, relative to the package root
```
Comment (piotmag769) [inline snforge_std/src/fs/file_operations.cairo:12]: #2011

Comment (Arcticae) [reply snforge_std/src/fs/file_operations.cairo:12]: why would we do that

File: docs/src/appendix/snforge-library/fs.md (hunk 7 above)
Comment (Arcticae) [reply docs/src/appendix/snforge-library/fs.md:29]: ah tomato tomato

**Pull request merged**